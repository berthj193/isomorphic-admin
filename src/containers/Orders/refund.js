import React from 'react';
import PropTypes from 'prop-types';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import withOrdersManager from '../../components/apolloProviders/withOrdersManager';
import RefundForm from './refundForm';
import { createRefund } from '../../graphql/orders';

export class CreateRefund extends React.Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
    mutate: PropTypes.func.isRequired,
    orders: PropTypes.arrayOf(PropTypes.object),
  };

  static defaultProps = {
    orders: [{}],
  };

  handleSubmit = values => {
    const { payment_id, amount_pence } = values;
    this.props.mutate({
      mutation: createRefund,
      variables: {
        payment_id,
        amount_pence: parseInt(amount_pence, 0),
      },
      messages: ['Refund added successfully'],
    });
  };

  render() {
    const order = this.props.orders[0];

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Orders</h1>
          <ButtonList buttons={[{ link: `/dashboard/orders/${order.id}`, title: 'Go back' }]} />
          <RefundForm
            onSubmit={this.handleSubmit}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withOrdersManager(CreateRefund);
