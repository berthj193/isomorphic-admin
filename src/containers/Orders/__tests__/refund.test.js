import React from 'react';
import { CreateRefund } from '../refund';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<OrderCreateRefund />', () => {
  const getComponent = (props = { mutate: () => null }) => shallow(<CreateRefund {...props} />);
  const mutate = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();

    mutate.mockReturnValue(Promise.resolve({ foo: 'bar' }));
  });

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should execute mutate function on submit', () => {
    const wrapper = getComponent({ mutate });
    wrapper.instance().handleSubmit({});
    expect(mutate).toBeCalled();
  });
});
