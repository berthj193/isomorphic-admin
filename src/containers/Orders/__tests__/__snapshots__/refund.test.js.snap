// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderCreateRefund /> should render properly 1`] = `
<LayoutWrapper
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <LayoutContent>
    <h1>
      Orders
    </h1>
    <ButtonList
      buttons={
        Array [
          Object {
            "link": "/dashboard/orders/undefined",
            "title": "Go back",
          },
        ]
      }
    />
    <ReduxFormWrapper
      onSubmit={[Function]}
    />
  </LayoutContent>
</LayoutWrapper>
`;
